src/sas/sascalc/data_util/nxsunit.py

```diff
@@ -191,16 +189,30 @@ def _format_unit_structure(unit=None):
     :param unit: Unit string to be formatted
     :return: Formatted unit string
     """
-    if unit:
-        # Convert a/b^n to a*b^{-n}
-        split = unit.split("/")
-        if len(split) == 1:
-            return unit
-        else:
-            split_ct = split[1].split("^")
-            number = 1 if len(split_ct) == 1 else split_ct[1]
-            split[0] = '' if split[0] == '1' else split[0] + '*'
-            return "{0}{1}^{{-{2}}}".format(split[0], split_ct[0], number)
+    if not unit:
+        return
+    # a-m[ /?]b-n ... -> a^m b^-n and pre_Unit -> preUnit
+    unit = re.sub('([℃ÅA-Za-z ]+)([-0-9]+)', r"\1^\2", unit).replace("_", '')
+    # invUnit -> /unit
+    unit = re.sub('inv', '/', unit, flags=re.IGNORECASE)
+    # Capture multi-unit
+    split_ws = unit.split()
+    final = ''
+    for item in split_ws:
+        # a/b^n, a*inv{b^n} or a*b^n -> a*b^{[-?]n}
+        split = item.split("/")
+        index = 1 if len(split) > 1 else 0
+        split_ct = split[index].split("^")
+        # Break if unit format is prefixUnit
+        if len(split_ct) == 1 and not index:
+            final += "{0} ".format(item)
+            break
+        number = 1 if len(split_ct) == 1 else split_ct[1]
+        start = '' if split[0] in ['1', '', item] else split[0] + '*'
+        sign = '-' if len(split_ct) == 1 else ''
+        final += "{0}{1}^{{{2}{3}}} ".format(start, split_ct[0], sign, number)
+    # Remove leading and trailing whitespace and double brackets with single
+    return final.strip().replace('{{', '{').replace('}}', '}')
```

Comment:
Can you end up with `a^{b^{c}}` with the double close brace eaten?

Do you ever end up with double braces?

I wonder what `a/b/c` should do, and if it does it right?